pennylane/qchem/openfermion_obs.py

```diff
@@ -942,14 +967,29 @@ def molecular_hamiltonian(
         core, active = qml.qchem.active_space(
             mol.n_electrons, mol.n_orbitals, mult, active_electrons, active_orbitals
         )
-        if args is None:
-            h = qml.qchem.diff_hamiltonian(mol, core=core, active=active)()
-            coeffs = qml.numpy.real(h.coeffs, requires_grad=False)
-            h = qml.Hamiltonian(coeffs, h.ops, grouping_type=grouping_type, method=grouping_method)
+
+        need_grad = args is not None
```

Comment:
For some reason, `need_grad` sounds very funny.

Reply:
replaced with `requires_grad`